pyee/util.py

```diff
@@ -0,0 +1,38 @@
+# -*- coding: utf-8 -*-
+
+from collections import OrderedDict
+
+
+def evolve(cls, underlying, *args, **kwargs):
+    """A helper to create instances of an event emitter ``cls`` that inherit
+    event behavior from an ``underlying`` event emitter instance.
+
+    When called, this instantiates a new instance of ``cls`` and then copies
+    all the handlers on ``underlying`` to that new instance, before returning
+    it. Afterwards, any modifications made to the underlying event emitter
+    and the new "evolved" event emitter are not shared.
+
+    This is mostly helpful if you have a simple underlying event emitter
+    with event handlers already attached to it, but you want to use that
```

Comment:
I think it's equally likely that the underlying library will /emit/ events on `underlying`. Some ideas:

* Write a procedure that decorates the instance method `underlying.on` so that the event is actually emitted on `new`
* Write a procedure that decorates the instance method `underlying.on` so that the event is *additionally* emitted on 'new'

With the former, this handler-copying business seems fine, but with the latter there's no need.

Reply:
The former also would want to account for detecting new listeners on underlying and copying them over